java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -113,20 +113,30 @@ private static native Dataset createWithFfiStream(long arrowStreamMemoryAddress,
    * @return Dataset
    */
   public static Dataset open(String path, BufferAllocator allocator) {
+    return open(path, allocator, new DatasetOpenOptions());
+  }
+
+  /**
+   * Open a dataset from the specified path with additional options.
+   *
+   * @param path      file path
+   * @param allocator Arrow buffer allocator.
+   * @param options   the open options
+   * @return Dataset
+   */
+  public static Dataset open(String path, BufferAllocator allocator, DatasetOpenOptions options) {
```

Comment:
Can we have a few override functions, some with the simplest API possible?

i.e.

```
public static Dataset open(String uri);
public static Dataset open(String, options);
public static Dataset open(String, options, allocator);

```

Based on the possibility of which signature is likely to be used more.

Also rename `DatasetOpenOptions` to `OpenOptions` or `ReadOptions` to be consistent with the rust API?